wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -448,3 +486,39 @@ private class SourceCodeHelper(
     return m
   }
 }
+
+private class SourceInfo(
+  val helper: SourceCodeHelper,
+  path: List<Int> = emptyList(),
+) {
+  constructor(
+    fileDescriptor: FileDescriptorProto,
+    path: List<Int> = emptyList()
+  ) : this(SourceCodeHelper(fileDescriptor), path)
+
+  private val path = mutableListOf(*path.toTypedArray())
+
+  fun push(value: Int) {
+    path.add(value)
+  }
+
+  fun info(): LocationAndComments {
+    return helper.getLocation(path)
+  }
+
+  fun new(): SourceInfo {
```

Comment:
I think a method like this is usually named `clone` or `copy`.

Reply:
👍